plugins/samples/cdn_token_generator/plugin.cc

```diff
@@ -0,0 +1,234 @@
+      size_t m = json.find('"', l + 1);
+      if (l != std::string::npos && m != std::string::npos) {
+        std::string val = json.substr(l + 1, m - l - 1);
+        kv[key] = val;
+        i = m + 1;
+      } else {
+        break;
+      }
+    }
+    if (kv.count("privateKeyHex")) config_.privateKeyHex = kv["privateKeyHex"];
+    if (kv.count("keyName")) config_.keyName = kv["keyName"];
+    if (kv.count("expirySeconds")) config_.expirySeconds = std::stoi(kv["expirySeconds"]);
+    if (kv.count("urlHeaderName")) config_.urlHeaderName = kv["urlHeaderName"];
+    if (kv.count("outputHeaderName")) config_.outputHeaderName = kv["outputHeaderName"];
+    return !config_.privateKeyHex.empty() && !config_.keyName.empty()
+        && !config_.urlHeaderName.empty() && !config_.outputHeaderName.empty();
+  }
+};
+
+class CDNTokenHttpContext : public Context {
+ public:
+  explicit CDNTokenHttpContext(uint32_t id, RootContext* root)
+      : Context(id, root),
+        config_(static_cast<CDNTokenRootContext*>(root)->config()) {}
+
+  // Función para loguear sin prefijos de archivo/línea
+  void logExact(const std::string& message) {
+    logInfo(message);  // Función de bajo nivel sin prefijos
+  }
+
+  FilterHeadersStatus onRequestHeaders(uint32_t, bool) override {
+    if (config_.privateKeyHex.empty()) {
+      logExact("Plugin configuration is null");
+      return FilterHeadersStatus::Continue;
+    }
+    auto originalURLPtr = getRequestHeader(config_.urlHeaderName);
+    std::string originalURL = originalURLPtr ? originalURLPtr->toString() : "";
+    if (originalURL.empty()) {
+      logExact("URL header not found or empty: " + config_.urlHeaderName);
+      return FilterHeadersStatus::Continue;
+    }
+    // Validate + sign URL
+    if (!validateURL(originalURL)) {
+      logExact("Invalid URL provided");
+      return FilterHeadersStatus::Continue;
+    }
+    logExact("Generating signed URL for: " + originalURL);
+
+    std::string signedURL = generateSignedURL(originalURL);
+    if (signedURL.empty()) {
+      logExact("Failed to generate signed URL");
+      return FilterHeadersStatus::Continue;
+    }
+    addRequestHeader(config_.outputHeaderName, signedURL);
+    return FilterHeadersStatus::Continue;
+  }
+
+ private:
+  Config config_;
+
+  bool validateURL(const std::string& url) {
+    if (url.size() > MAX_URL_LENGTH) return false;
+    // Regex similar to Go
+    std::regex url_pattern(R"(^https?://[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,}(/[^\s]*)?$)");
+    if (!std::regex_match(url, url_pattern)) return false;
+    // Basic internal host block
+    std::string lower = url;
+    std::transform(lower.begin(), lower.end(), lower.begin(), ::tolower);
+    if (lower.find("localhost") != std::string::npos ||
+        lower.find("127.0.0.1") != std::string::npos ||
+        lower.find("::1") != std::string::npos) {
+      return false;
+    }
+    return true;
+  }
+
+  std::vector<uint8_t> hexToBytes(const std::string& hex) {
```

Comment:
Use an Abseil function, like absl::HexStringToBytes(), or absl::Base64Unescape (if you decide to store the private key as base64)